### PR TITLE
[RTM] Adds request and response stacking of header merging

### PR DESCRIPTION
**Pull request by aschempp**
This fixes issues when fragments are rendered inline using subrequests. Otherwise native headers are not pushed to the final response.

### CODE REVIEW

File: src/EventListener/MergeHttpHeadersListener.php [hunk 1]
```diff
@@ -11,13 +11,16 @@
 namespace Contao\CoreBundle\EventListener;
 
 use Contao\CoreBundle\Framework\ContaoFrameworkInterface;
+use Contao\CoreBundle\HttpKernel\Header\HeaderStorageInterface;
+use Contao\CoreBundle\HttpKernel\Header\NativeHeaderStorage;
 use Symfony\Component\HttpFoundation\Response;
```
Comment (leofeyer) [inline src/EventListener/MergeHttpHeadersListener.php]: `HttpFoundation` comes before `HttpKernel`. 😄

Comment (aschempp) [reply src/EventListener/MergeHttpHeadersListener.php]: Yeah, but Symfony does not come before Contao 😉

Comment (leofeyer) [reply src/EventListener/MergeHttpHeadersListener.php]: LOL

File: tests/EventListener/MergeHttpHeadersListenerTest.php [hunk 2]
```diff
@@ -227,7 +228,10 @@ public function testHeadersAreInheritedFromSubrequests()
         $this->assertSame('application/json', $response->headers->get('Content-Type'));
     }
 
-    public function testMultiHeadersAreInheritedFromSubrequests()
+    /**
+     * Tests that mulit headers are inherited from a subrequest.
```
Comment (bytehead) [inline tests/EventListener/MergeHttpHeadersListenerTest.php]: Typo: `mulit` instead of `multi`.

Comment (aschempp) [reply tests/EventListener/MergeHttpHeadersListenerTest.php]: Fixed in 5d4b63390db09308b27c9447d27a687aa35cd0f3

**Comment by leofeyer** [comment]
Thank you @aschempp.

**Comment by bytehead** [comment]
🎉

**Pull request merged**